plugins/filter_aws/aws.c

```diff
@@ -237,21 +230,76 @@ static int get_metadata(struct flb_filter_aws *ctx, char *metadata_path,
         return -1;
     }
 
-    tmp = flb_sds_create_len(client->resp.payload, client->resp.payload_size);
+    if (key != NULL) {
+        /* get the value of the key from payload json string */
+        tmp = flb_json_get_val(client->resp.payload,
+                               client->resp.payload_size, key);
+        if (!tmp) {
+            tmp = flb_sds_create_len("NULL", 4);
+            flb_plg_info(ctx->ins,
+                         "ec2-metadata: %s is undefined in EC2 instance", key);
```

Comment:
Probably should be warn or error instead

Reply:
Also you don't need to prefix it with "ec2-metadata:". The flb_plg_* functions will print the plugin name in all log messages

Reply:
Changed to error and removed the prefix of `"ec2-metadata:"`